File: app/src/main/java/com/chess/cryptobot/model/room/CryptoBotDatabase.java
package com.chess.cryptobot.model.room;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import com.chess.cryptobot.model.room.converter.Converters;

@Database(version = 2, entities = {ProfitPair.class, BtcBalance.class}, exportSchema = false)
@TypeConverters(Converters.class)
public abstract class CryptoBotDatabase extends RoomDatabase {
    private static CryptoBotDatabase database;

    public abstract ProfitPairDao getProfitPairDao();

    public abstract BtcBalanceDao getBtcBalanceDao();

    public static CryptoBotDatabase getInstance(Context context) {
        if (database == null) {
            database = Room.databaseBuilder(context, CryptoBotDatabase.class, "cryptobotDB")
                    .enableMultiInstanceInvalidation()
                    .fallbackToDestructiveMigration()
                    .build();
        }
        return database;
    }
}

File: app/src/main/java/com/chess/cryptobot/model/room/CryptoBotDatabase.kt
package com.chess.cryptobot.model.room

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import com.chess.cryptobot.model.room.converter.TimestampConverter
import com.chess.cryptobot.util.SingletonHolder

@Database(version = 7, entities = [
    ProfitPair::class,
    CryptoBalance::class,
    BalanceSyncTicker::class,
    PairMinTradeSize::class,
    CoinInfo::class
], exportSchema = false)
@TypeConverters(TimestampConverter::class)
abstract class CryptoBotDatabase : RoomDatabase() {
    abstract val profitPairDao: ProfitPairDao
    abstract val cryptoBalanceDao: CryptoBalanceDao
    abstract val balanceSyncDao: BalanceSyncDao
    abstract val minTradeSizeDao: PairMinTradeSizeDao
    abstract val coinInfoDao: CoinInfoDao

    companion object : SingletonHolder<CryptoBotDatabase, Context>({
        Room.databaseBuilder(it.applicationContext, CryptoBotDatabase::class.java, "cryptobotDB")
                .enableMultiInstanceInvalidation()
                .fallbackToDestructiveMigration()
                .build()
    })
}